commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/CommonsLangPortedFractionTest.java

```diff
@@ -0,0 +1,1148 @@
+ *      assertThrows(ArithmeticException.class, () -> Fraction.of(Integer.MAX_VALUE, 1, 2));
+ *      assertThrows(ArithmeticException.class, () -> Fraction.of(-Integer.MAX_VALUE, 1, 2));
+ *
+ *      // very large
+ *      f = Fraction.of(-1, 0, Integer.MAX_VALUE);
+ *      assertEquals(-1, f.getNumerator());
+ *      assertEquals(1, f.getDenominator());
+ *
+ *      // negative denominators not allowed in this constructor.
+ *      assertThrows(ArithmeticException.class, () -> Fraction.of(0, 4, Integer.MIN_VALUE));
+ *      assertThrows(ArithmeticException.class, () -> Fraction.of(1, 1, Integer.MAX_VALUE));
+ *      assertThrows(ArithmeticException.class, () -> Fraction.of(-1, 2, Integer.MAX_VALUE));
+ *  }
+ */
+
+    @Test
+    public void testReducedFactory_int_int() {
+        Fraction f = null;
+
+        // zero
+        f = Fraction.of(0, 1);
+        assertEquals(0, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+
+        // normal
+        f = Fraction.of(1, 1);
+        assertEquals(1, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+
+        f = Fraction.of(2, 1);
+        assertEquals(2, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+
+        // improper
+        f = Fraction.of(22, 7);
+        assertEquals(22, f.getNumerator());
+        assertEquals(7, f.getDenominator());
+
+        // negatives
+        f = Fraction.of(-6, 10);
+        assertEquals(-3, f.getNumerator());
+        assertEquals(5, f.getDenominator());
+
+        f = Fraction.of(6, -10);
+        assertEquals(3, f.getNumerator());
+        assertEquals(-5, f.getDenominator());
+
+        f = Fraction.of(-6, -10);
+        assertEquals(-3, f.getNumerator());
+        assertEquals(-5, f.getDenominator());
+
+        // zero denominator
+        assertThrows(ArithmeticException.class, () -> Fraction.of(1, 0));
+        assertThrows(ArithmeticException.class, () -> Fraction.of(2, 0));
+        assertThrows(ArithmeticException.class, () -> Fraction.of(-3, 0));
+
+        // reduced
+        f = Fraction.of(0, 2);
+        assertEquals(0, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+
+        f = Fraction.of(2, 2);
+        assertEquals(1, f.getNumerator());
+        assertEquals(1, f.getDenominator());
+
+        f = Fraction.of(2, 4);
+        assertEquals(1, f.getNumerator());
+        assertEquals(2, f.getDenominator());
+
+        f = Fraction.of(15, 10);
+        assertEquals(3, f.getNumerator());
+        assertEquals(2, f.getDenominator());
+
+        f = Fraction.of(121, 22);
+        assertEquals(11, f.getNumerator());
+        assertEquals(2, f.getDenominator());
+
+        // Extreme values
+        // OK, can reduce before negating
+        f = Fraction.of(-2, Integer.MIN_VALUE);
+        assertEquals(-1, f.getNumerator());
+        assertEquals(Integer.MIN_VALUE / 2, f.getDenominator());
+
+        /*
+         *  This is allowed in [numbers]
```

Comment:
For cases that are allowed in numbers and not lang I think the original assertion from lang should be commented out and replaced with new assertions. This would be:
```java
// lang requires the sign to be in the numerator so this would throw.
// numbers allows the sign to be in the denominator so this does not throw.
// assertThrows(ArithmeticException.class, () -> Fraction.getReducedFraction(-7, Integer.MIN_VALUE));
f = Fraction.of(-7, Integer.MIN_VALUE);
assertEquals(1, f.signum());
assertEquals(-7, f.getNumerator());
assertEquals(Integer.MIN_VALUE, f.getDenominator());
```

Reply:
@aherbert done.